### DISCUSSION TITLE
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!

### DISCUSSION THREAD

**Discussion by dast96** [bulletphysics/bullet3#3432] [General]
For Some Reason i get this error, trying to run a pybullet world with `python run.py` with the demo world of pycram see https://github.com/cram2/pycram/tree/master/demos/pycram_pr2_bullet_world_demo
> python run.py 
> pybullet build time: Oct 13 2020 11:37:16
> 2020-10-13 22:19:18.466 Python[3695:1755969] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
> *** First throw call stack:
> (
> 	0   CoreFoundation                      0x00007fff3394cb57 __exceptionPreprocess + 250
> 	1   libobjc.A.dylib                     0x00007fff6c7985bf objc_exception_throw + 48
> 	2   CoreFoundation                      0x00007fff3397534c -[NSException raise] + 9
> 	3   AppKit                              0x00007fff30b6f5ec -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 310
> 	4   AppKit                              0x00007fff30b57052 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
> 	5   AppKit                              0x00007fff30b56ac3 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
> 	6   pybullet.cpython-38-darwin.so       0x000000010145ffbc Mac_createWindow + 892
> 	7   pybullet.cpython-38-darwin.so       0x000000010144f632 _ZN15MacOpenGLWindow12createWindowERK25b3gWindowConstructionInfo + 66
> 	8   pybullet.cpython-38-darwin.so       0x000000010132e20f _ZN16SimpleOpenGL3AppC2EPKciibiiii + 655
> 	9   pybullet.cpython-38-darwin.so       0x0000000101317677 _ZN20OpenGLExampleBrowser4initEiPPc + 919
> 	10  pybullet.cpython-38-darwin.so       0x0000000101187c92 _Z41btCreateInProcessExampleBrowserMainThreadiPPcb + 210
> 	11  pybullet.cpython-38-darwin.so       0x00000001011c0244 _ZN44InProcessPhysicsClientSharedMemoryMainThreadC2EiPPcb + 164
> 	12  pybullet.cpython-38-darwin.so       0x00000001011bfc00 b3CreateInProcessPhysicsServerAndConnectMainThread + 48
> 	13  pybullet.cpython-38-darwin.so       0x0000000100fad608 pybullet_connectPhysicsServer + 536
> 	14  Python                              0x00000001006ccf8e cfunction_call_varargs + 171
> 	15  Python                              0x00000001006cca7d _PyObject_MakeTpCall + 274
> 	16  Python                              0x000000010076d847 call_function + 804
> 	17  Python                              0x000000010076a257 _PyEval_EvalFrameDefault + 29861
> 	18  Python                              0x00000001006cd2c2 function_code_fastcall + 106
> 	19  Python                              0x000000010076d67d call_function + 346
> 	20  Python                              0x000000010076a23b _PyEval_EvalFrameDefault + 29833
> 	21  Python                              0x00000001006cd2c2 function_code_fastcall + 106
> 	22  Python                              0x000000010076d67d call_function + 346
> 	23  Python                              0x000000010076a23b _PyEval_EvalFrameDefault + 29833
> 	24  Python                              0x00000001006cd2c2 function_code_fastcall + 106
> 	25  Python                              0x00000001006cf3f0 method_vectorcall + 256
> 	26  Python                              0x00000001006ccd09 PyVectorcall_Call + 108
> 	27  Python                              0x00000001007e3465 t_bootstrap + 74
> 	28  Python                              0x00000001007a4ced pythread_wrapper + 25
> 	29  libsystem_pthread.dylib             0x00007fff6db45109 _pthread_start + 148
> 	30  libsystem_pthread.dylib             0x00007fff6db40b8b thread_start + 15
> )
> libc++abi.dylib: terminating with uncaught exception of type NSException
> [1]    3695 abort      /usr/local/bin/python3

Python version 3.8.6
pip version 20.2.3
pybullet version 3.0.6
macOS 10.15.7

any Ideas?

**Comment by erwincoumans** [comment]
Thanks for the report. The most important part of information is missing: what world? and what command-line did you use to cause the crash?

**Comment by dast96** [comment]
> Thanks for the report. The most important part of information is missing: what world? and what command-line did you use to cause the crash?

Sorry, its happens when i use `python run.py` or `python3 run.py` i have an alias so i doenst matter if i use python or python3

The pycram bullet world demo
https://github.com/cram2/pycram/tree/master/demos/pycram_pr2_bullet_world_demo

**Comment by dast96** [comment]
i tried it aswell with 3.7.9 - because i have an notebook with debian and it only works with 3.7.9 there..
any ideas...? it seems it has something to do with the world?

 "Terminating app due to uncaught exception 'NSInternalInconsistencyException'" - i dont know where i have to catch this exception to get a better debug information.
